internal/juju/kubernetes_clouds.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/api/client/cloud"
+	k8s "github.com/juju/juju/caas/kubernetes"
+	k8scloud "github.com/juju/juju/caas/kubernetes/cloud"
+	"k8s.io/client-go/tools/clientcmd"
+	"strings"
+)
+
+type kubernetesCloudsClient struct {
+	SharedClient
+}
+
+type CreateKubernetesCloudInput struct {
+	Name                  string
+	KubernetesContextName string
+	KubernetesConfig      string
+	ParentCloudName       string
+	ParentCloudRegion     string
+}
+
+type ReadKubernetesCloudInput struct {
+	Name             string
+	KubernetesConfig string
```

Comment:
question: why is KubernetesConfig necessary to read the cloud info from juju?

Reply:
agreed, this is the rudiment, I will drop it.